packages/x-charts/package.json

```diff
@@ -72,27 +68,10 @@
     "@mui/internal-test-utils": "^1.0.3",
     "@react-spring/core": "^9.7.3",
     "@react-spring/shared": "^9.7.3",
-    "@types/d3-color": "^3.1.3",
-    "@types/d3-delaunay": "^6.0.4",
-    "@types/d3-interpolate": "^3.0.4",
-    "@types/d3-scale": "^4.0.8",
-    "@types/d3-shape": "^3.1.6",
     "@types/prop-types": "^15.7.12",
     "csstype": "^3.1.3",
     "rimraf": "^5.0.7"
   },
-  "exports": {
-    ".": {
-      "types": "./index.d.ts",
-      "import": "./esm/index.js",
-      "default": "./esm/index.js"
-    },
-    "./*": {
-      "types": "./*/index.d.ts",
-      "import": "./esm/*/index.js",
-      "default": "./esm/*/index.js"
-    }
-  },
```

Comment:
Why this was removed? 🤔 

It is probably the reason it is not working in node without the `index.js` https://nodejs.org/api/esm.html#mandatory-file-extensions

Reply:
We probably also want to put the `"./package.json": "./package.json"` in this list 😅

Reply:
> Why this was removed? 🤔

I got the answer: Otherwise, NextJs still needs to transpile `@mui/x-charts`

Those lines kind of fake the package is ESM only, but it's not yet done

Here are nice explanation of what's missing from @Janpot 

> The only drawback I see is that we're not really addressing the root cause of the issue. Our intention was to be a esm only package with x-charts, but our execution is not 100%. i.e.
> - we still have some commonjs sources in our package folder
> - we still have package.json main field pointing to a commonjs source
> - our internal imports are missing .js extension or /index.js
> - we don't put type: 'module' in package.json (or use other signals to tell node how to execute the source, e.g. .mjs)
>
> This confuses Next.js serverside code, because it's technically correct in doing what we signal it to do, which is "execute us as commonjs".

The solution in this PR makes the CommonJS build working, so we should drop those lines such that package managers use the CJS one.


In brief, what happens is

1. The CJS build is broken because it needs to do some ESM imports for d3-*
2. We modify the package.json such that everybody uses only the ESM build
3. Since we only provide ESM users need to transpile the package

This PR uses a trick to fix 1, those lines revert 2, to avoid having to request 3 from users

I realized that when doing the test on issue reproductions again:

- Next application with those lines: https://codesandbox.io/p/devbox/eager-rain-szymx2?file=%2Fnext.config.js%3A8%2C3
- Next application after removing those lines: https://codesandbox.io/p/devbox/elegant-forest-hvn8fk?file=%2Fnext.config.js%3A7%2C42&workspaceId=d214a4af-c616-40eb-ac12-807102cf00db

Reply:
Got it, thanks for the explanation